backend/controllers/users.js

```diff
@@ -115,27 +115,33 @@ const editPost = async (req,res,next)=>{
         cover:filename
         
     })
-        res.json(postDoc)
+        res.status(200).json(postDoc)
     } catch (error) {
         next(error)
     }
 }
 
 const deletePost =async(req,res,next)=>{
 
-    const {id} = req.body
+    const {id} = req.params
     const {token} = req.cookies
     try {
```

Comment:
Here, instead of wrapping each route's code with try/catch, you can use a wrapper function that calls next automatically for you when an error is thrown.
Check this link https://ioannisioannou.me/using-async-await-in-express-js/